website/server/controllers/api-v4/inbox.js

```diff
@@ -147,4 +147,97 @@ api.getInboxMessages = {
+ * @apiName FlagPrivateMessage
+ * @apiGroup Member
+ *
+ * @apiParam (Path) {UUID} messageId The private message id
+ *
+ * @apiSuccess {Object} data The flagged private message
+ * @apiSuccess {UUID} data.id The id of the message
+ * @apiSuccess {String} data.text The text of the message
+ * @apiSuccess {Number} data.timestamp The timestamp of the message in milliseconds
+ * @apiSuccess {Object} data.likes The likes of the message (always an empty object)
+ * @apiSuccess {Object} data.flags The flags of the message
+ * @apiSuccess {Number} data.flagCount The number of flags the message has
+ * @apiSuccess {UUID} data.uuid The User ID of the author of the message,
+ *                              or of the recipient if `sent` is true
+ * @apiSuccess {String} data.user The Display Name of the author of the message,
+ *                                or of the recipient if `sent` is true
+ * @apiSuccess {String} data.username The Username of the author of the message,
+ *                                    or of the recipient if `sent` is true
+ *
+ * @apiUse MessageNotFound
+ * @apiUse MessageIdRequired
+ * @apiError (400) {BadRequest} messageGroupChatFlagAlreadyReported You have already
+ *                                                                  reported this message
+ */
+api.flagPrivateMessage = {
+  method: 'POST',
+  url: '/members/flag-private-message/:messageId',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    const chatReporter = chatReporterFactory('Inbox', req, res);
+    const message = await chatReporter.flag();
+    res.respond(200, {
+      ok: true,
+      message,
+    });
+  },
+};
+
+/**
+ * @api {post} /api/v4//inbox/like-private-message/:messageId Like a private message
+ * @apiName LikePrivateMessage
+ * @apiGroup Inbox
+ * @apiDescription Likes a private message
+ *
+ * @apiParam (Path) {UUID} messageId The private message id
+ *
+ * @apiSuccess {Object} data The liked <a href='https://github.com/HabitRPG/habitica/blob/develop/website/server/models/message.js#L42' target='_blank'>private message</a>
+ *
+ * @apiUse MessageNotFound
+ */
+api.likePrivateMessage = {
+  method: 'POST',
+  url: '/inbox/like-private-message/:messageId',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    req.checkParams('messageId', apiError('messageIdRequired')).notEmpty();
+
+    const validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    const { user } = res.locals;
+    const { messageId } = req.params;
+
+    const message = await inboxLib.getUserInboxMessage(user, messageId);
+    if (!message) throw new NotFound(res.t('messageGroupChatNotFound'));
+
+    // update like for the owners message (each message is saved for both users)
+    if (!message.likes) message.likes = {};
+    message.likes[user._id] = !message.likes[user._id];
+    message.markModified('likes');
+    await message.save();
+
+    // user message it themselves
```

Comment:
I don't know what this comment means :sweat_smile: Isn't the code beneath this all about when the message in question _isn't_ the user's own message?

Reply:
I'm trying to ask that myself as well ... :/ will remove or rewrite that one

Reply:
I think that condition needs to be removed at all - since otherwise the "liked" amount will be different on the opponent chat (if the current user liked their own message) will apply the tests for that as well